assignment/core/src/main/java/com/assignment/core/services/impl/CipherService.java

```diff
@@ -0,0 +1,35 @@
+package com.assignment.core.services.impl;
+
+import com.assignment.core.services.AsymmetricCipherProvider;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(service = CipherService.class)
+public class CipherService {
+
+    @Reference(target = "(algorithm=RSA)")
```

Comment:
Does this follow Open Closed principle ?
Is this Open for Extension and Closed for Modification ?